dev-tools/omdb/src/bin/omdb/mgs.rs

```diff
@@ -0,0 +1,499 @@
+        .await
+        .with_context(|| format!("fetching components for SP {:?}", sp_id));
+    let list = match component_list {
+        Ok(l) => l.into_inner(),
+        Err(e) => {
+            eprintln!("error: {:#}", e);
+            return Ok(());
+        }
+    };
+
+    #[derive(Tabled)]
+    #[tabled(rename_all = "SCREAMING_SNAKE_CASE")]
+    struct SpComponentRow<'a> {
+        name: &'a str,
+        description: &'a str,
+        device: &'a str,
+        presence: String,
+        serial: String,
+    }
+
+    impl<'a> From<&'a SpComponentInfo> for SpComponentRow<'a> {
+        fn from(v: &'a SpComponentInfo) -> Self {
+            SpComponentRow {
+                name: &v.component,
+                description: &v.description,
+                device: &v.device,
+                presence: format!("{:?}", v.presence),
+                serial: format!("{:?}", v.serial_number),
+            }
+        }
+    }
+
+    if list.components.is_empty() {
+        println!("    COMPONENTS: none found\n");
+        return Ok(());
+    }
+
+    let table_rows = list.components.iter().map(SpComponentRow::from);
+    let table = tabled::Table::new(table_rows)
+        .with(tabled::settings::Style::empty())
+        .with(tabled::settings::Padding::new(0, 1, 0, 0))
+        .to_string();
+    println!("    COMPONENTS\n");
+    println!("{}", textwrap::indent(&table.to_string(), "        "));
+    println!("");
+
+    #[derive(Tabled)]
+    #[tabled(rename_all = "SCREAMING_SNAKE_CASE")]
+    struct CabooseRow {
+        component: String,
+        board: String,
+        git_commit: String,
+        name: String,
+        version: String,
+    }
+
+    impl<'a> From<(&'a SpIdentifier, &'a SpComponentInfo, SpComponentCaboose)>
+        for CabooseRow
+    {
+        fn from(
+            (_sp_id, component, caboose): (
+                &'a SpIdentifier,
+                &'a SpComponentInfo,
+                SpComponentCaboose,
+            ),
+        ) -> Self {
+            CabooseRow {
+                component: component.component.clone(),
+                board: caboose.board,
+                git_commit: caboose.git_commit,
+                name: caboose.name,
+                version: caboose.version.unwrap_or_else(|| "-".to_string()),
+            }
+        }
+    }
+
+    let mut cabooses = Vec::new();
+    for c in &list.components {
+        if !COMPONENTS_WITH_CABOOSES.contains(&c.component.as_str()) {
+            continue;
+        }
+
+        for i in 0..1 {
+            let r = mgs_client
+                .sp_component_caboose_get(
+                    sp_id.type_,
+                    sp_id.slot,
+                    &c.component,
+                    i,
+                )
+                .await
+                .with_context(|| {
+                    format!(
+                        "get caboose for sp type {:?} sp slot {} \
+                                component {:?} slot {}",
```

Comment:
Tiny nit - some extra whitespace here, probably from `cargo fmt` not realigning `\`-broken strings